src/scripts/model/master.js

```diff
@@ -101,8 +98,8 @@ class MasterModel extends EventEmitter {
         const { opts, parent } = this;
         const { path, enablePolling, interval } = opts;
 
-        if (parent && parent.log instanceof Function) {
-            parent.log(`Fetching data from path: ${path}`);
+        if (opts.log instanceof Function) {
```

Comment:
Should be moved to the constructor to avoid this check every time we want to log something in the model. We provide `opts.log` by default, I'm assuming this is an additional safety check in case a developer passes in an invalid type.

```js
this.log = this.opts.log;
```

```js
if (!(this.log instanceof Function)) {
    throw new Error('Invalid dependency in initialization: log should be a Function');
}
```